lib/api/core/hotelClerk.js

```diff
@@ -555,17 +555,18 @@ class HotelClerk {
     }
     else {
       room.customers.delete(requestContext.connection.id);
+    }
 
-      const request = new Request({
-        controller: 'realtime',
-        action: 'unsubscribe',
-        index: room.index,
-        collection: room.collection,
-        volatile
-      }, requestContext);
+    // even if the room is deleted for this node, another one may need the notification
```

Comment:
Ok I had a _very_ hard time following what's going on here while in cluster mode, so please correct me if I'm wrong.

I have an issue about that room removal function.
This is not exactly about that line in particular, but that entire function bothers me, and here is why: whenever a user executes the `realtime:subscribe` API route, the cluster plugin makes other nodes artificially create an empty Koncorde room.
If I had to guess, I would say that it's to allow the `realtime:join` route to work.

But what this bugfix shows is that a room can be destroyed on one node, while still existing on other ones. Which, in other terms, means that we have a desync.

I'm not quite sure that this should be resolved in this one PR. But if I had to work on it, I would remove the artifical room creation made by the cluster plugin, and rely on the redis status to check whether a room can be joined or not.

What do you think?

Reply:
I need to check again, but I think we still need a room in Koncorde on every node to check if a document write matches a possible subscription.

Regarding the room deletion, we do not sync the hotelclerk and on that level, it is fine. We still keep the room in Koncorde if needed by using an ugly hack in the cluster: https://github.com/kuzzleio/kuzzle-plugin-cluster/blob/master/lib/index.js#L270 though.

Still, I agree, I am not very happy with the way the cluster behaves for the time being. I would tend to think that the next move would be to include it in Kuzzle core. This way, we could get rid of this kind of ugly overrides and make things a little clearer.

Reply:
Ohh I missed that hack. And it's ugly all right. :expressionless: 

I agree, integrating the cluster plugin directly into the core is the right move, probably to be done for the v2 though :+1: